### PR TITLE
Update for OSQP 0.4.0

### PR DESCRIPTION
**Pull request by bstellato**
Update for the new release https://github.com/oxfordcontrol/osqp/releases/tag/v0.4.0

In particular:

- Handle the new status for nonconvex problem detected `OSQP_NON_CVX`
- Adapt interface to use [qdldl](https://github.com/oxfordcontrol/qdldl/) instead of suitesparse ldl

### CODE REVIEW

File: src/MathOptInterfaceOSQP.jl [hunk 1]
```diff
@@ -531,6 +531,8 @@ function MOI.get(optimizer::OSQPOptimizer, a::MOI.TerminationStatus)
         MOI.AlmostSuccess
     elseif osqpstatus == :Primal_infeasible_inaccurate
         MOI.AlmostSuccess
+    elseif osqpstatus == :Non_convex
+	MOI.InvalidModel
```
Comment (mlubin) [inline src/MathOptInterfaceOSQP.jl]: Yep, this is the right status code to use.
Note the inconsistent indentation.

File: src/OSQP.jl [hunk 2]
```diff
@@ -26,7 +26,7 @@ function __init__()
 
     depsdir = realpath(joinpath(dirname(@__FILE__), "..", "deps"))
     if (vnum.major != 0 && vnum.minor != 2)
```
Comment (mlubin) [inline src/OSQP.jl]: vnum.minor needs to change to 4? Seems like this should fail as is.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+Version 0.2.0 (23 July 2018): OSQP v0.4.0
+-----------------------------------------
+* Upgraded OSQP to 0.4.0 version
+
```
Comment (tkoolen) [inline CHANGELOG.md]: Ah, sorry for not keeping this up to date.

**Comment by rdeits** [comment]
I would not suggest using a `git clone` in the Julia build process. It could certainly be made to work for the Julia build, but it's going to be a bit slower, and, more importantly, it doesn't solve the more fundamental problem that anyone *else* who downloads the sources from your Github release will have the same problem. 

Instead, I'd suggest making sure the github releases actually contain the full source tree (It's really unfortunate that submodules don't work in this situation). I think it should be possible for you to clone the project, do a `git submodule update --init`, zip up the result, and then attach that zip file to your github release, as in https://blog.github.com/2013-07-02-release-your-software/#creating-releases

Alternatively, you could use [`git subtree`](https://www.atlassian.com/blog/git/alternatives-to-git-submodule-git-subtree) instead of a submodule to manage the QDLDL sources inside OSQP. I've found `git subtree` harder to work with than a submodule, but it has the advantage that the subtree files are actually included in the superproject, so GitHub releases would just work. 

Or, frankly, you could just copy-paste the QDLDL sources directly into OSQP, and just copy-paste in a new version whenever you want to update it. A bit inelegant, but would certainly solve the problem.

**Comment by bstellato** [comment]
Thank you @mlubin and @rdeits for the suggestions. 

For the moment, I have decided to download the qdldl sources during the build process. Github releases do not have an official command line interface to automate that process and I am a bit reluctant to set up some custom clients via CI. 

Probably we will add the automatic sources clone + zip + upload to bintray.com for future releases [as we do for the precompiled libraries](https://bintray.com/bstellato/generic/OSQP/0.4.0#files/OSQP/0.4.0).

**Comment by bstellato** [comment]
Tests are passing now. If you guys are happy I can merge and tag a new release.

**Comment by rdeits** [comment]
> For the moment, I have decided to download the qdldl sources during the build process. Github releases do not have an official command line interface to automate that process and I am a bit reluctant to set up some custom clients via CI.

Yup, perfectly reasonable. Thank you!

**Pull request merged**